vcstool/clients/git.py

```diff
@@ -497,6 +561,23 @@ def _get_remote_urls(self):
             'returncode': 0 if remote_urls else 1
         }
 
+    def _get_sparse_patterns(self):
+        cmd = [GitClient._executable, 'sparse-checkout', 'list']
+        result = self._run_command(cmd)
+        if result['returncode']:
+            result['output'] = 'Could not list sparse patterns: ' \
```

Comment:
Not sure if this user-facing message, but if it is, then don't we want to give more hints for why?

Looking at what `git sparse-checkout list` returns, I guess this `if` block hits when the repo is not set for `sparse-checkout`?

```
$ git sparse-checkout list
warning: this worktree is not sparse (sparse-checkout file may not exist)

$ git sparse-checkout set pkg_c
$ ll
total 16K
drwxr-xr-x 4 n130s n130s 4.0K Jun 16 13:57 ./
drwxr-xr-x 8 n130s n130s 4.0K Jun 16 13:57 .git/
drwxrwxr-x 4 n130s n130s 4.0K Jun 16 13:57 pkg_c/
drwxrwxr-x 3 n130s n130s 4.0K Jun 16 13:57 ../
$ git sparse-checkout list
pkg_c
```

Reply:
The pattern I saw elsewhere is to list the previous commands' output following the diagnostic. Looks like I omitted the key, but it's there now.

This also brings up an interesting point on having a sparse-checkout without any patterns. The command will return with 0. I suppose this could be a valid use case if you only want the top level of a repository, so I'll rewrite to ensure we don't fail with an empty pattern list.